src/tests/util.rs

```diff
@@ -511,3 +511,106 @@ pub fn generate_key_pair(
     }
     Ok((pub_key, pri_key))
 }
+
+struct CheckTemplate<'a> {
+    bytes: Vec<(CK_ATTRIBUTE_TYPE, Vec<u8>, &'a [u8])>,
+    longs: Vec<(CK_ATTRIBUTE_TYPE, CK_ULONG, CK_ULONG)>,
+    bools: Vec<(CK_ATTRIBUTE_TYPE, CK_BBOOL, CK_BBOOL)>,
+    tmpl: Vec<CK_ATTRIBUTE>,
+}
+
+pub fn check_attributes(
+    session: CK_SESSION_HANDLE,
+    handle: CK_OBJECT_HANDLE,
+    ulongs: &[(CK_ATTRIBUTE_TYPE, CK_ULONG)],
+    bools: &[(CK_ATTRIBUTE_TYPE, bool)],
+    bytes: &[(CK_ATTRIBUTE_TYPE, &[u8])],
+) -> Option<String> {
+    let capacity = ulongs.len() + bytes.len() + bools.len();
+    let mut template = CheckTemplate {
+        bytes: Vec::with_capacity(bytes.len()),
+        longs: Vec::with_capacity(ulongs.len()),
+        bools: Vec::with_capacity(bools.len()),
```

Comment:
The `CheckTemplate` arguments sorts them yet in another order so I would find any usage of this API (even though its in tests and internal) almost impossible from memory without checking the actual order over and over again.